communication/comms.py

```diff
@@ -0,0 +1,185 @@
+"""This module contains a Comms class for providing client-server communication between two pokemon."""
+
+import asyncio
+import random
+import pokemon.pokemon as pkmn
+
+
+class Comms(pkmn.ElectricPokemon):
+    """
+    This Comms class inherits from the ElectricPokemon class.
+
+    Adds client-server communication between pokemon, implemented with asyncio and co-routines.
+
+    Attributes
+    ----------
+    name : str
+        Name of pokemon.
+    nickname : str
+        Nickname of pokemon.
+    moves : list
+        A list of 4 pokemon moves.
+    pokemon_type : str
+        Type of pokemon.
+    ip_address : str
+        The IP address as the client or server (default 127.0.0.1).
+    port_no : int
+        The port number to communicate on (default 8765).
+    """
+
+    def __init__(self, name, nickname, pokemon_type, ip_address="127.0.0.1", port_no=8765):
+        """Initialise Comms class object.
+
+        Initialise with the arguments name, nickname and type, and with
+        optional keyward arguments ip_address and port_no.
+
+        Parameters
+        ----------
+        name : str
+            Name of pokemon.
+        nickname : str
+            Nickname of pokemon.
+        pokemon_type : str
+            Type of pokemon.
+        ip_address : str
+            The IP address as the client or server (default 127.0.0.1).
+        port_no : int
+            The port number to communicate on (default 8765).
+        """
+        self.ip_address = ip_address
+        self.port_no = port_no
+        super(Comms, self).__init__(name, nickname, pokemon_type)
+
+    async def msg_tx(self, message, writer):
+        """Co-routine writes message to socket.
+
+        Parameters
+        ----------
+        message : str
+            The message to write to socket.
+        writer :  StreamWriter object
+            Object instance of StreamWriter class for writing to socket.
+        """
+        print(f"Sent: {message!r}")
+        writer.write(message.encode())
+        await writer.drain()
+
+    async def list_tasks(self, writer):  # noqa: D301
+        """Co-routine create a random number of tasks for writing messages to socket.
+
+        Each message ends with a \n. Multiple messages are sent as a group
+        of messages, and gets terminated with 'EOM' to indicate the end of transmission.
+
+        Parameters
+        ----------
+        writer :  StreamWriter object
+            Object instance of StreamWriter class for writing to socket.
+        """
+        message = self.name + ".\n"
+        tasks = []
+        random_no = random.randint(1, 5)
+        print(f"Random number generated = {random_no}")
+        for i in range(random_no):
+            tasks.append(self.msg_tx(message, writer))
+        await asyncio.gather(*tasks)
+        await self.msg_tx("EOM\n", writer)
+
+    async def read_data(self, reader):
+        """Co-routine reads from socket.
```

Comment:
```suggestion
        """Read from socket.
```

If you really want to make *more* explicit that it's a co-routine (though `async def` does this just fine already), then you can include the adverb "asynchronously" in your docstring, perhaps like this:

```suggestion
        """Read asynchronously from socket.
```